website/docs/cdktf/concepts/best-practices.mdx

```diff
@@ -0,0 +1,30 @@
+---
+page_title: "Best Practices"
+description: "We discuss patterns and strategies for building and maintaining infrastructure with CDKTF."
+---
+
+# Best Practices
+
+## Use `TerraformVariable` to read secrets
+
+If you read a secret from the environment or out of a file with normal file system access and you use them in your CDKTF code, they will end up in the synthesized Terraform code. The problem with that is that it might leak if an attacker can get access to your Terraform code. To mitigate this, you should [use the `TerraformVariable` construct](/cdktf/concepts/variables-and-outputs#input-variables) to read secrets.
```

Comment:
```suggestion
If you read a secret from the environment or out of a file with normal file system access and you use it in your CDKTF code, it will end up in the synthesized Terraform code. The problem with that is that it might leak if an attacker can get access to your Terraform code. To mitigate this, you should [use the `TerraformVariable` construct](/cdktf/concepts/variables-and-outputs#input-variables) to read secrets.
```

Reply:
The Terraform variable construct won't "read" secrets – maybe phrase that as "to pass secrets"?

Maybe also mention TFC VCS integration and the need to commit generated TF code in which case secrets would be committed to source control.